src/components/KTable/KTable.vue

```diff
@@ -868,6 +868,13 @@ const initData = () => {
   hasInitialized.value = true
 }
 
+// Ensure `props.headers` are reactive.
+watch(() => props.headers as TableHeader[], (newVal: TableHeader[]) => {
```

Comment:
suggestion: rather than force-casting the props, could we add the proper interface within the `defineProps` macro?

```typescript
headers: {
  type: Array as PropType<TableHeader[]>,
  default: () => [],
},
```

Reply:
We'll need to ensure that existing usage doesn't choke on the type change (e.g. pull the package preview into `konnect-ui-apps` in a draft PR and run the `typecheck` command recursively)

Reply:
https://github.com/Kong/konnect-ui-apps/pull/3608 -- I tweaked the table header type to match how it's actually used in the wild, and that seems to work.